### ISSUE TITLE
Server Error - Invalid Endpoint and Secret

### ISSUE THREAD

**Issue by awaisiqbal01** [bigbluebutton/greenlight#5356]
I have come across an issue with Bigbluebutton on a server with just around 20 users. The system started displaying the error message, "Server Error, Invalid endpoints and secret," on certain accounts. Upon further investigation, I noticed that these particular accounts had a significant number of recordings associated with them. I am seeking guidance on the possible reasons behind this issue and the appropriate solution. Thank you for your assistance.
Bigbluebutton version: 2.4.9
greenlight version:  2.12.6
os: ubuntu18
Please let me know if you require any additional information to address this matter effectively.

**Comment by JeanPluzo** [comment]
Hi,
have you checked your certificates? If this started to happen suddenly, perhaps your TLS certificate has expired.
OTOH it would be wise to update BBB to 2.6 and GL to 2.14 in the near future.

Regards,
J.

**Comment by awaisiqbal01** [comment]
Thank you for your response. If the issue were related to certificates, it shouldn't work with new accounts. However, when I create a new account on the same server, it does not give me the error.

**Comment by awaisiqbal01** [comment]
Here are the logs
`2023-08-07 07:59:39 +0000 - INFO: [1ec3aceb-9d81-46bf-8316-a6d4b297c9b5] [172.21.0.1] show:  show 
2023-08-07 07:59:42 +0000 - ERROR: [1ec3aceb-9d81-46bf-8316-a6d4b297c9b5] [172.21.0.1] BigBlueButtonException: Connection error. Your URL is probably incorrect: "https://test2.onescreensolutions.com/bigbluebutton/api". Error: Failed to open TCP connection to test2.onescreensolutions.com:443 (Host is unreachable - connect(2) for "test2.onescreensolutions.com" port 443), messageKey: IncorrectUrlError 
2023-08-07 07:59:42 +0000 - ERROR: [1ec3aceb-9d81-46bf-8316-a6d4b297c9b5] [172.21.0.1] BigBlueButtonException: **********m  Manual error handling /usr/src/app/vendor/bundle/ruby/2.7.0/gems/bigbluebutton-api-ruby-1.9.0/lib/bigbluebutton_api.rb:781:in `rescue in send_request'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/bigbluebutton-api-ruby-1.9.0/lib/bigbluebutton_api.rb:755:in `send_request'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/bigbluebutton-api-ruby-1.9.0/lib/bigbluebutton_api.rb:719:in `send_api_request'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/bigbluebutton-api-ruby-1.9.0/lib/bigbluebutton_api.rb:222:in `is_meeting_running?'
/usr/src/app/app/controllers/concerns/bbb_server.rb:29:in `room_running?'
/usr/src/app/app/controllers/rooms_controller.rb:83:in `show'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_controller/metal/basic_implicit_render.rb:6:in `send_action'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/abstract_controller/base.rb:194:in `process_action'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_controller/metal/rendering.rb:30:in `process_action'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/abstract_controller/callbacks.rb:42:in `block in process_action'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.7.1/lib/active_support/callbacks.rb:132:in `run_callbacks'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/abstract_controller/callbacks.rb:41:in `process_action'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_controller/metal/rescue.rb:22:in `process_action'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_controller/metal/instrumentation.rb:34:in `block in process_action'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.7.1/lib/active_support/notifications.rb:168:in `block in instrument'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.7.1/lib/active_support/notifications/instrumenter.rb:23:in `instrument'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.7.1/lib/active_support/notifications.rb:168:in `instrument'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_controller/metal/instrumentation.rb:32:in `process_action'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_controller/metal/params_wrapper.rb:256:in `process_action'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/activerecord-5.2.7.1/lib/active_record/railties/controller_runtime.rb:24:in `process_action'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/abstract_controller/base.rb:134:in `process'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionview-5.2.7.1/lib/action_view/rendering.rb:32:in `process'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_controller/metal.rb:191:in `dispatch'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_controller/metal.rb:252:in `dispatch'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/routing/route_set.rb:52:in `dispatch'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/routing/route_set.rb:34:in `serve'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/journey/router.rb:52:in `block in serve'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/journey/router.rb:35:in `each'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/journey/router.rb:35:in `serve'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/routing/route_set.rb:840:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/omniauth-2.1.0/lib/omniauth/strategy.rb:202:in `call!'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/omniauth-2.1.0/lib/omniauth/strategy.rb:169:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/omniauth-2.1.0/lib/omniauth/strategy.rb:202:in `call!'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/omniauth-2.1.0/lib/omniauth/strategy.rb:169:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/omniauth-2.1.0/lib/omniauth/builder.rb:44:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/http_accept_language-2.1.1/lib/http_accept_language/middleware.rb:14:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3.1/lib/rack/tempfile_reaper.rb:15:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3.1/lib/rack/etag.rb:27:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3.1/lib/rack/conditional_get.rb:27:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3.1/lib/rack/head.rb:12:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/http/content_security_policy.rb:18:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3.1/lib/rack/session/abstract/id.rb:266:in `context'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3.1/lib/rack/session/abstract/id.rb:260:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/middleware/cookies.rb:670:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/middleware/callbacks.rb:28:in `block in call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.7.1/lib/active_support/callbacks.rb:98:in `run_callbacks'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/middleware/callbacks.rb:26:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/middleware/debug_exceptions.rb:61:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/lograge-0.11.2/lib/lograge/rails_ext/rack/logger.rb:15:in `call_app'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/railties-5.2.7.1/lib/rails/rack/logger.rb:26:in `block in call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.7.1/lib/active_support/tagged_logging.rb:71:in `block in tagged'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.7.1/lib/active_support/tagged_logging.rb:28:in `tagged'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.7.1/lib/active_support/tagged_logging.rb:71:in `tagged'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/railties-5.2.7.1/lib/rails/rack/logger.rb:26:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/request_store-1.5.1/lib/request_store/middleware.rb:19:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/middleware/request_id.rb:27:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3.1/lib/rack/method_override.rb:24:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3.1/lib/rack/runtime.rb:22:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/middleware/executor.rb:14:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/middleware/static.rb:127:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3.1/lib/rack/sendfile.rb:110:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/actionpack-5.2.7.1/lib/action_dispatch/middleware/ssl.rb:74:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/railties-5.2.7.1/lib/rails/engine.rb:524:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3.1/lib/rack/urlmap.rb:74:in `block in call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3.1/lib/rack/urlmap.rb:58:in `each'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3.1/lib/rack/urlmap.rb:58:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/puma-4.3.12/lib/puma/configuration.rb:228:in `call'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/puma-4.3.12/lib/puma/server.rb:727:in `handle_request'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/puma-4.3.12/lib/puma/server.rb:476:in `process_client'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/puma-4.3.12/lib/puma/server.rb:332:in `block in run'
/usr/src/app/vendor/bundle/ruby/2.7.0/gems/puma-4.3.12/lib/puma/thread_pool.rb:134:in `block in spawn_thread' 
2023-08-07 07:59:42 +0000 - INFO: [1ec3aceb-9d81-46bf-8316-a6d4b297c9b5] [172.21.0.1] method=GET path=/b/sha-rmp-mv7-ic1 format=html controller=RoomsController action=show status=200 duration=3159.13 view=39.61 db=11.12 host=greenlight `

**Comment by JeanPluzo** [comment]
Hi,
you're using a private ip address. This is not recommended.
Either get a public ip address or user your system's DNS settings (hosts file, etc) to route in the right direction.
It might be that your hosts in the private address range can connect to the BBB server, but the server itself might not know it.

Regards,
J.

**Comment by awaisiqbal01** [comment]
Thank you J. for your reply here are the more logs when fetching recording list
`2023-08-09 08:25:42 +0000 - INFO: [9a64e1c4-f637-41cc-b15d-90bf523316e9] [115.186.136.159] show:  show 
2023-08-09 08:25:42 +0000 - ERROR: [9a64e1c4-f637-41cc-b15d-90bf523316e9] [115.186.136.159] Show: awa-rld-hvk-blc start failed: {} 
2023-08-09 08:25:42 +0000 - INFO: [9a64e1c4-f637-41cc-b15d-90bf523316e9] [115.186.136.159] Indise elseif condition  
2023-08-09 08:25:42 +0000 - INFO: [9a64e1c4-f637-41cc-b15d-90bf523316e9] [115.186.136.159] Support: ordered_rooms ordered_rooms #<Room::ActiveRecord_Associations_CollectionProxy:0x00007f090b63c3e0> 
2023-08-09 08:25:42 +0000 - INFO: [9a64e1c4-f637-41cc-b15d-90bf523316e9] [115.186.136.159] @room_running = false => lecture#2 params[:name] 
2023-08-09 08:25:50 +0000 - INFO: [582c509e-02bb-40ac-8db8-5163f99f0bd4] [115.186.136.159] lesson lecture room count 78 
2023-08-09 08:25:50 +0000 - INFO: [582c509e-02bb-40ac-8db8-5163f99f0bd4] [115.186.136.159] lesson lecture room count 78 Lesson Recording(s) 
2023-08-09 08:25:50 +0000 - ERROR: [582c509e-02bb-40ac-8db8-5163f99f0bd4] [115.186.136.159] room_bbb_id, search_params = {}, ret_search_params = false jjxl7b92ztjzodrrw4qv9o0cnn1czd5wsoabgn9h 
2023-08-09 08:25:51 +0000 - INFO: [9a64e1c4-f637-41cc-b15d-90bf523316e9] [115.186.136.159] lesson lecture room count 78 
2023-08-09 08:25:51 +0000 - INFO: [9a64e1c4-f637-41cc-b15d-90bf523316e9] [115.186.136.159] lesson lecture room count 78 Lesson Recording(s) 
2023-08-09 08:25:51 +0000 - ERROR: [9a64e1c4-f637-41cc-b15d-90bf523316e9] [115.186.136.159] room_bbb_id, search_params = {}, ret_search_params = false jjxl7b92ztjzodrrw4qv9o0cnn1czd5wsoabgn9h 
2023-08-09 08:25:52 +0000 - INFO: [582c509e-02bb-40ac-8db8-5163f99f0bd4] [115.186.136.159] show:  can_create_rooms 2  
2023-08-09 08:25:52 +0000 - ERROR: [582c509e-02bb-40ac-8db8-5163f99f0bd4] [115.186.136.159] @paggy syntax ----- #<Pagy:0x00007f090be6e0b8> 
2023-08-09 08:25:52 +0000 - INFO: [582c509e-02bb-40ac-8db8-5163f99f0bd4] [115.186.136.159] Support: ordered_rooms ordered_rooms #<Room::ActiveRecord_Associations_CollectionProxy:0x00007f09135cbac0> 
2023-08-09 08:25:52 +0000 - INFO: [582c509e-02bb-40ac-8db8-5163f99f0bd4] [115.186.136.159] method=GET path=/b/awa-rld-hvk-blc format=html controller=RoomsController action=show status=200 duration=20700.61 view=65.26 db=12.75 host=greenlight 
2023-08-09 08:25:53 +0000 - INFO: [9a64e1c4-f637-41cc-b15d-90bf523316e9] [115.186.136.159] show:  can_create_rooms 2  
2023-08-09 08:25:53 +0000 - ERROR: [9a64e1c4-f637-41cc-b15d-90bf523316e9] [115.186.136.159] @paggy syntax ----- #<Pagy:0x00007f091096d898> 
2023-08-09 08:25:53 +0000 - INFO: [9a64e1c4-f637-41cc-b15d-90bf523316e9] [115.186.136.159] Support: ordered_rooms ordered_rooms #<Room::ActiveRecord_Associations_CollectionProxy:0x00007f090b63c3e0> 
2023-08-09 08:25:53 +0000 - INFO: [9a64e1c4-f637-41cc-b15d-90bf523316e9] [115.186.136.159] method=GET path=/b/awa-rld-hvk-blc format=html controller=RoomsController action=show status=200 duration=10566.49 view=58.58 db=7.54 host=greenlight 
2023-08-09 08:25:55 +0000 - INFO: [781d2601-0661-488a-aa78-76d6b327dd00] [115.186.136.159] Support: ordered_rooms ordered_rooms #<Room::ActiveRecord_Associations_CollectionProxy:0x00007f090b3552a8> 
2023-08-09 08:25:55 +0000 - ERROR: [781d2601-0661-488a-aa78-76d6b327dd00] [115.186.136.159] recordings params {"controller"=>"users", "action"=>"recordings", "user_uid"=>"gl-epajxhsgebfg"}  recordings array =["jjxl7b92ztjzodrrw4qv9o0cnn1czd5wsoabgn9h", "2vzk5estqy0db6opuvwtdqbs8sz40hcgouuztn2d"],,,,, current_user.rooms.pluck(:bbb_id)=>>>["jjxl7b92ztjzodrrw4qv9o0cnn1czd5wsoabgn9h", "2vzk5estqy0db6opuvwtdqbs8sz40hcgouuztn2d"] 
2023-08-09 08:25:55 +0000 - INFO: [781d2601-0661-488a-aa78-76d6b327dd00] [115.186.136.159] get_multiple_recordings: ["jjxl7b92ztjzodrrw4qv9o0cnn1czd5wsoabgn9h", "2vzk5estqy0db6opuvwtdqbs8sz40hcgouuztn2d"] get_multiple_recordings 
2023-08-09 08:26:15 +0000 - ERROR: [781d2601-0661-488a-aa78-76d6b327dd00] [115.186.136.159] BigBlueButtonException: Timeout error. Your server is probably down: "https://staging5.onescreensolutions.com/bigbluebutton/api". Error: Net::ReadTimeout, messageKey: TimeoutError 
2023-08-09 08:26:15 +0000 - INFO: [781d2601-0661-488a-aa78-76d6b327dd00] [115.186.136.159] method=GET path=/b/u/gl-epajxhsgebfg/recordings format=html controller=UsersController action=recordings status=200 duration=20083.57 view=24.63 db=3.73 host=greenlight `

**Comment by awaisiqbal01** [comment]
when number of recordings increases 
`http://yourserver.com/bigbluebutton/api/getRecordings?[parameters]&checksum=[checksum]`

 this API takes time to respond and this error happens. If I call same API from postman it works fine but in app it crashes

**Comment by hostbbb** [comment]
this is v2 GL and 2.4BBB no longer supported.  I believe you may be running into an issue similar to what wordpress encountered.  If the getRecording is using GET and the number of recordings appending to the query string exceeded i think like 1024,  the call fails.   Just guessing.   look in /var/log/nginx/bigbluebutton.access.log at the getRecordingcall and see the response.  if you have lots of recordings may be the issue.

**Comment by awaisiqbal01** [comment]
I have identified the issue: when the number of recordings increases, the BBB API takes longer to send a response. The default timeout in the BigBlueButtonApi is set to 10 seconds. When a request takes longer than 10 seconds, this error occurs.

I attempted to address this by modifying the default timeout in the gem file using the following code:
`bbb_server.instance_variable_set(:@timeout, 70);` //default timeout set to 70 sec
While this change made it work, I am aware that it's not the appropriate solution. It appears that as the number of recordings continues to grow, this approach will likely fail as well. Therefore, a proper and more robust solution is needed.
